src/main/java/org/jetbrains/research/refactorinsight/actions/ToggleLabelsAction.java

```diff
@@ -0,0 +1,22 @@
+package org.jetbrains.research.refactorinsight.actions;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.ToggleAction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.research.refactorinsight.services.WindowService;
+
+/**
+ * Toggle labeling of commits containing refactorings in VCSTable.
+ */
+public class ToggleLabelsAction extends ToggleAction {
```

Comment:
Why not use `BooleanPropertyToggleAction` or `DumbAwareToggleAction`?

Reply:
Good call, changed it to dumbaware